nodes-lib/src/codex.ts

```diff
@@ -141,16 +141,45 @@ const backfillNewStream = async (
   return streamID;
 };
 
+/**
+ * Get full historical publish state of a research object.
+*/
+export const getFullState = async (
+  streamID: string,
+) => {
+  const ceramic = newCeramicClient(getNodesLibInternalConfig().ceramicNodeUrl);
```

Comment:
why create a new client each time? is it worth to perhaps have option to pass a client in so we can reuse it? not sure the cost of creating a client, if it's low perhaps not an issue

Reply:
Fair point but seems cheap so I'll avoid breakage rn :angel: